src/main/java/duckling/Configuration.java

```diff
@@ -1,18 +1,39 @@
 package duckling;
 
 import duckling.errors.BadArgumentsError;
+import duckling.requests.Request;
 import duckling.routing.RouteDefinitions;
 import duckling.routing.Routes;
 
-import java.io.FileInputStream;
-import java.io.FileNotFoundException;
-import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
 
 public class Configuration {
     public static final int DEFAULT_PORT = 5000;
     public static final String DEFAULT_ROOT = ".";
     public static final RouteDefinitions DEFAULT_ROUTES = new RouteDefinitions(
+        Routes.get("/parameters").with((Request request) -> {
+            try {
+                ArrayList<String> lines = new ArrayList<>();
+
+                for (String line: request.getQuery().split("&")) {
+                    ArrayList<String> items = new ArrayList<>();
+                    String decoded = URLDecoder.decode(line, "utf-8");
+
+                    Collections.addAll(items, decoded.split("=", 2));
+
+                    lines.add(items.stream().collect(Collectors.joining(" = ")));
+                }
+
+                return lines.stream().collect(Collectors.joining(Server.CRLF));
+            } catch (UnsupportedEncodingException exception) {
+                return "Unable to parse request URL query params.";
+            }
+        }),
```

Comment:
I bet you can find a way to break this up - it's getting pretty long and takes awhile to comprehend.